libs/ui-lib/lib/ocm/components/AddHosts/day2Wizard/Day2ClusterDetails.tsx

```diff
@@ -104,6 +107,41 @@ const Day2ClusterDetails = () => {
     [day2Cluster.id, wizardContext],
   );
 
+  const onChangeCpuArchitectureDropdown = React.useCallback(
+    (value: string, initialValues: Day2ClusterDetailValues): void => {
+      if (value !== initialValues.cpuArchitecture) {
+        //Fetch infraEnv by clusterId and cpu architecture
+        void InfraEnvsService.getInfraEnv(day2Cluster.id, value as SupportedCpuArchitecture)
+          .then((infraEnv) => {
+            if (!infraEnv && pullSecret) {
+              //If infraEnv don't exist, create a new one
+              void InfraEnvsService.create({
+                name: `${day2Cluster.name || ''}_infra-env-${value}`,
+                pullSecret,
+                clusterId: day2Cluster.id,
+                openshiftVersion: day2Cluster.openshiftVersion,
+                cpuArchitecture: value as ClusterCpuArchitecture,
+              });
+            }
+          })
+          .catch(() => {
+            if (pullSecret) {
+              //If infraEnv don't exist, create a new one
+              void InfraEnvsService.create({
+                name: `${day2Cluster.name || ''}_infra-env-${value}`,
+                pullSecret,
+                clusterId: day2Cluster.id,
+                openshiftVersion: day2Cluster.openshiftVersion,
+                cpuArchitecture: value as ClusterCpuArchitecture,
+              });
```

Comment:
This is exactly the same code as above, try extracting this code into a reusable function please.